pandas/tests/test_common.py

```diff
@@ -213,3 +213,20 @@ def test_frozenlist(self):
         result = df[frozen]
         expected = df[[]]
         tm.assert_frame_equal(result, expected)
+
+
+@pytest.mark.parametrize("with_exception", [True, False])
+def test_temp_setattr(with_exception):
+    # GH#45954
+    ser = Series(dtype=object)
+    ser.name = "first"
+    try:
```

Comment:
Maybe a bit more robust to validate the error raised:

```
if with_exception:
    maybe_raises = pytest.raises(ValueError, match="Exception raised")
else:
    maybe_raises = contextlib.nullcontext()

with maybe_raises:
    with com.temp_setattr(ser, "name", "second"):
        ...
    
```

Reply:
Good call! flake8 didn't like using pytest.raises outside a context manager; raising in either case made it a little simpler.